src/software/multithreading/thread_safe_buffer.hpp

```diff
@@ -151,8 +151,9 @@ void ThreadSafeBuffer<T>::push(const T& value)
     std::scoped_lock<std::mutex> buffer_lock(buffer_mutex);
     if (log_buffer_full && buffer.full())
     {
-        LOG(WARNING) << "Pushing to a full ThreadSafeBuffer of type: " << TYPENAME(T)
-                     << std::endl;
```

Comment:
This causes a circular dependency in Bazel (`ProtoLogger` depends on `ThreadSafeBuffer`, but `ProtoLogger` is used by the Logger). I wasn't sure how to fix it other than removing the log (will probably replace it with `std::cout`).